ARExperiment/ARExperiment/LightsAnimationsViewController.swift

```diff
@@ -64,20 +68,17 @@ class LightsAnimationsViewController: UIViewController {
     }
 
     private func addNodeToSessionUsingFeaturePoints(location: CGPoint) {
-        let hitResultsFeaturePoints: [ARHitTestResult] =
-            sceneView.hitTest(location, types: .featurePoint)
-
-        if let hit = hitResultsFeaturePoints.first {
-            let rotate = simd_float4x4(SCNMatrix4MakeRotation(sceneView.session.currentFrame!.camera.eulerAngles.y, 0, 1, 0))
-            let finalTransform = simd_mul(hit.worldTransform, rotate)
-            let anchor = ARAnchor(transform: finalTransform)
-            sceneView.session.add(anchor: anchor)
+        guard let hitTransfrom = arModel.getTransformForAnchor(location: location,
```

Comment:
might want to think about naming here but :100: for extracting.

Be careful that, whilst you're extracting stuff into the ViewModel, that doesn't just get massive instead :D you can take a look in there and see if you see multiple levels of abstraction :)

Reply:
Will take a look 😄 So far I have only used it for specific things we need to repeat on every view controller to calculate ARView related things. Because I was having issues where one controller worked correctly and the others did not because the code was not updated. 

If some logic does not have to do with AR Scene it should not go to the view model. But honestly if I was doing a proper refactor I would have created my own `SCNNode` and one controller for all the cases and my own `ARSCNView` so that way the functionality could be properly split and always the same for all the demos.